source/Ship.cpp

```diff
@@ -2401,6 +2401,15 @@ bool Ship::FireAntiMissile(const Projectile &projectile, vector<Visual> &visuals
 
 
 
+bool Ship::IsInAntiMissileRangeOf(const Projectile &projectile)
+{
+	if(projectile.Position().Distance(position) > antiMissileRange)
+		return false;
+	return true;
```

Comment:
If instead of returning a boolean, you return a magic value (e.g. `-1`) for not-in-range, and the distance(squared) between the ship and the projectile when in range, we can reduce the number of distance calculations that need to be done in the new algorithm.